### ISSUE TITLE
Support for `using`

### ISSUE THREAD

**Issue by vchuravy** [JuliaInterop/Clang.jl#430]
```
/// These callbacks are used to return multiple types from functions while
/// transferring ownership to the caller. The first argument is the number of
/// consecutive elements pointed to by the second argument. The third argument
/// is an opaque pointer forwarded to the callback by the caller.
using MlirTypesCallback = void (*)(intptr_t, MlirType *, void *);
```

Lead to:
```
/home/vchuravy/.julia/artifacts/c90711842814121dacf839a59bca4c40f573cbff/include/mlir-c/Interfaces.h:51:1: error: unknown type name 'using'
/home/vchuravy/.julia/artifacts/c90711842814121dacf839a59bca4c40f573cbff/include/mlir-c/Interfaces.h:51:27: error: expected expression
/home/vchuravy/.julia/artifacts/c90711842814121dacf839a59bca4c40f573cbff/include/mlir-c/Interfaces.h:60:45: error: unknown type name 'MlirTypesCallback'
```

**Comment by Gnimuc** [comment]
`using MlirTypesCallback = void (*)(intptr_t, MlirType *, void *);`

it looks like this is C++.

**Comment by vchuravy** [comment]
I see I will need to grumble at upstream for putting that into a C header

**Comment by vchuravy** [comment]
Is there a way to work around this in Clang.jl?

**Comment by Gnimuc** [comment]
closed by #480, #432, #435
